gapic/ads-templates/%namespace/%name/%version/%sub/services/%service/client.py.j2

```diff
@@ -806,14 +808,7 @@ class {{ service.client_name }}(metaclass={{ service.client_name }}Meta):
     {% endif %}
 
 
-try:
-    DEFAULT_CLIENT_INFO = gapic_v1.client_info.ClientInfo(
-        gapic_version=pkg_resources.get_distribution(
-            "{{ api.naming.warehouse_package_name }}",
-        ).version,
-    )
-except pkg_resources.DistributionNotFound:
-    DEFAULT_CLIENT_INFO = gapic_v1.client_info.ClientInfo()
+DEFAULT_CLIENT_INFO = gapic_v1.client_info.ClientInfo(gapic_version=package_version.__version__)
```

Comment:
It looks like this is grabbing the version that's defined in the above `gapic_version` file, is that correct? 

The implementation is actually to access the version of the Ads client library itself, see the logic [here](https://github.com/googleads/google-ads-python/blob/main/google/ads/googleads/v15/services/services/google_ads_service/client.py#L4467)

If you look at go/ads-client-libs-user-agent-design you can see that we opted to use the `gccl` user agent k/v for our own logging purposes, which is why the `gapic_version` keyword arg is used here instead of something else. 

Is it possible to retrieve the client library version without pkg_resources? If not, then we might need to add that logic in our interceptors before this change can be made.

Reply:
> It looks like this is grabbing the version that's defined in the above gapic_version file, is that correct?

Yes, that is correct. In [google-cloud-python](https://github.com/googleapis/google-cloud-python), we use [release-please](https://github.com/googleapis/release-please) to update the version in `gapic_version.py` prior to releasing. As an example see https://github.com/googleapis/google-cloud-python/pull/12010.

> The implementation is actually to access the version of the Ads client library itself, see the logic [here](https://github.com/googleads/google-ads-python/blob/main/google/ads/googleads/v15/services/services/google_ads_service/client.py#L4467)

We've moved away from this in GAPIC, and chose to populate `gapic_version.py` instead of a hardcoded version number in setup.py. Customers can obtain the version number programatically using `__version__`.

>Is it possible to retrieve the client library version without pkg_resources? If not, then we might need to add that logic in our interceptors before this change can be made.

It is possible to use something else like https://docs.python.org/3/library/importlib.metadata.html but we chose to have a version module instead.